src/components/app/routes/data/utils.js

```diff
@@ -269,50 +262,104 @@ export async function ensureAuthenticatedUser(requestUrl, params) {
  */
 export async function ensureActiveEnterpriseCustomerUser({
   enterpriseSlug,
-  activeEnterpriseCustomer,
-  staffEnterpriseCustomer,
-  allLinkedEnterpriseCustomerUsers,
+  enterpriseLearnerData,
+  isBFFData,
   requestUrl,
+  authenticatedUser,
+  queryClient,
 }) {
+  let {
+    enterpriseCustomer,
+    activeEnterpriseCustomer,
+    allLinkedEnterpriseCustomerUsers,
+  } = enterpriseLearnerData;
+  const {
+    shouldUpdateActiveEnterpriseCustomerUser,
+  } = enterpriseLearnerData;
+  const matchedBFFQuery = resolveBFFQuery(requestUrl.pathname);
   // If the enterprise slug in the URL matches the active enterprise customer user's slug OR no
   // active enterprise customer exists, return early.
-  if (!activeEnterpriseCustomer || activeEnterpriseCustomer.slug === enterpriseSlug) {
-    return null;
+  let nextActiveEnterpriseCustomer = null;
+
+  if (shouldUpdateActiveEnterpriseCustomerUser) {
+    // If this flag is truthy, we already know that the active enterprise customer user should be updated.
+    nextActiveEnterpriseCustomer = enterpriseCustomer;
+  } else if (!isBFFData) {
+    // Otherwise, if we're using the non-BFF API data, we must determine if the active
+    // enterprise customer user should be updated based on the enterprise slug in the URL.
+
+    // If the enterprise slug in the URL matches the active enterprise customer user's
+    // slug OR no active enterprise customer exists, return early.
+    if (activeEnterpriseCustomer?.slug === enterpriseSlug) {
+      return {
+        enterpriseCustomer,
+        allLinkedEnterpriseCustomerUsers,
+      };
+    }
+    // Else, try to find the enterprise customer for the given slug and, if found, update it
+    // as the active enterprise customer for the learner.
+    const foundEnterpriseCustomerUserForSlug = allLinkedEnterpriseCustomerUsers.find(
+      enterpriseCustomerUser => {
+        if (!enterpriseCustomerUser.enterpriseCustomer) {
+          return false;
```

Comment:
[nit/clarification] When won't `enterpriseCustomerUser.enterpriseCustomer` exists during this iteration of `allLinkedEnterpriseCustomerUsers`? Was this check added due to bad test input data? I'm not sure there's a scenario where `/api/v1/enterprise-learner/` returns `{ enterprise_customer: null }` where this check would be needed? 🤔